contrib/examples/kafka/src/main/java/com/google/cloud/dataflow/contrib/kafka/examples/TopHashtagsExample.java

```diff
@@ -0,0 +1,257 @@
+            long tsMillis = jsonMapper.readTree(kv.getValue()).path("timestamp_ms").asLong();
+            return tsMillis == 0 ? Instant.now() : new Instant(tsMillis);
+          } catch (Exception e) {
+            throw Throwables.propagate(e);
+          }
+        }
+      };
+
+  // return json string containing top hashtags and window information time
+  private static class OutputFormatter extends DoFn<List<KV<String, Long>>, String>
+      implements DoFn.RequiresWindowAccess {
+
+    private transient DateTimeFormatter formatter;
+    private transient ObjectWriter jsonWriter;
+
+    static class OutputJson {
+      @JsonProperty String windowStart;
+      @JsonProperty String windowEnd;
+      @JsonProperty String generatedAt;
+      @JsonProperty List<HashtagInfo> topHashtags;
+
+      OutputJson(String windowStart, String windowEnd,
+                 String generatedAt, List<HashtagInfo> topHashtags) {
+        this.windowStart = windowStart;
+        this.windowEnd = windowEnd;
+        this.generatedAt = generatedAt;
+        this.topHashtags = topHashtags;
+      }
+    }
+
+    static class HashtagInfo {
+      @JsonProperty final String hashtag;
+      @JsonProperty final long count;
+      HashtagInfo(String hashtag, long count) {
+        this.hashtag = hashtag;
+        this.count = count;
+      }
+    }
+
+    @Override
+    public void processElement(ProcessContext ctx) throws Exception {
+      if (formatter == null) {
+        formatter = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss").withZoneUTC();
+        jsonWriter = new ObjectMapper().writerWithType(OutputJson.class);
+      }
+
+      List<HashtagInfo> topHashtags = new ArrayList<>(ctx.element().size());
+
+      for (KV<String, Long> tag : ctx.element()) {
+        topHashtags.add(new HashtagInfo(tag.getKey(), tag.getValue()));
+      }
+
+      IntervalWindow window = (IntervalWindow) ctx.window();
+
+      String json = jsonWriter.writeValueAsString(new OutputJson(
+          formatter.print(window.start()),
+          formatter.print(window.end()),
+          formatter.print(Instant.now()),
+          topHashtags));
+
+      ctx.output(json);
+    }
+  }
+
+  private static class KafkaWriter extends DoFn<String, Void> {
+
+    private final String topic;
+    private final Map<String, Object> config;
+    private transient KafkaProducer<String, String> producer = null;
+
+    public KafkaWriter(Options options) {
+      this.topic = options.getOutputTopic();
+      this.config = ImmutableMap.<String, Object>of(
+          "bootstrap.servers", options.getBootstrapServers(),
+          "key.serializer",    StringSerializer.class.getName(),
+          "value.serializer",  StringSerializer.class.getName());
+    }
+
+    @Override
+    public void processElement(ProcessContext ctx) throws Exception {
+      if (producer == null) {
+        producer = new KafkaProducer<String, String>(config);
+      }
+      LOG.info("Top Hashtags : " + ctx.element());
```

Comment:
Do not log per element -- super slow!

Just drop I think.

(If you were going to log, it would be `LOG.info("Top hashtags: {}", ctx.element());` -- for performance when logging is disabled.